pkg/server/event_handler.go

```diff
@@ -293,6 +317,44 @@ func (h *EventHandler[T, P]) getGCSObjectProto(ctx context.Context, objAttrs map
 	return p, nil
 }
 
+type payloadMetadata struct {
+	Metadata map[string]string `json:"metadata,omitempty"`
+}
+
+func parseGitHubSource(ctx context.Context, data []byte) (*v1alpha1.GitHubSource, error) {
+	var pm payloadMetadata
+	if err := json.Unmarshal(data, &pm); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal payloadMetadata %w", err)
+	}
+
+	r := &v1alpha1.GitHubSource{}
```

Comment:
Nit: It would be more idiomatic to do:

```go
var r v1alpha1.GitHubSource
```

since there are no fields being set on the struct during initialization.